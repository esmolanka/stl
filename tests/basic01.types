#eval Unit

#eval Void

#eval Unit -> Unit

#eval forall a. a -> a

#eval (forall a. a) -> (forall b. b)

#eval {}

#eval { foo : Unit }

#eval { foo? : Unit }

#eval forall (r : Row). { foo : Unit | r}

#eval exists (r : Row). { foo : Unit | r}

#eval <>

#eval < Foo : Unit >

