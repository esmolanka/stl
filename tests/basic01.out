tests/basic01.types:1:1-1:11:
  Normalised:
    Unit
  Kind:
    Type
tests/basic01.types:3:1-3:11:
  Normalised:
    Void
  Kind:
    Type
tests/basic01.types:5:1-5:19:
  Normalised:
    Unit -> Unit
  Kind:
    Type
tests/basic01.types:7:1-7:23:
  Normalised:
    ∀ a. a -> a
  Kind:
    Type
tests/basic01.types:9:1-9:36:
  Normalised:
    (∀ a. a) -> ∀ b. b
  Kind:
    Type
tests/basic01.types:11:1-11:9:
  Normalised:
    ∃ (ρ : Row). { | ρ}
  Kind:
    Type
tests/basic01.types:13:1-13:21:
  Normalised:
    ∃ (ρ : Row). {foo : Unit | ρ}
  Kind:
    Type
tests/basic01.types:15:1-15:22:
  Normalised:
    ∃ (ρ : Row). ∃ (ω : #). {foo^ω : Unit | ρ}
  Kind:
    Type
tests/basic01.types:17:1-17:42:
  Normalised:
    ∀ (r : Row). {foo : Unit | r}
  Kind:
    Type
tests/basic01.types:19:1-19:42:
  Normalised:
    ∃ (r : Row). {foo : Unit | r}
  Kind:
    Type
tests/basic01.types:21:1-21:9:
  Normalised:
    ∀ (ρ : Row). < | ... ρ>
  Kind:
    Type
tests/basic01.types:23:1-23:21:
  Normalised:
    ∀ (ρ : Row). ∃ (ω : #). <Foo^ω : Unit | ... ρ>
  Kind:
    Type